src/web/handlers.rs

```diff
@@ -376,3 +376,52 @@ pub fn put_bso(bso_req: BsoPutRequest) -> impl Future<Item = HttpResponse, Error
 pub fn get_configuration(creq: ConfigRequest) -> impl Future<Item = HttpResponse, Error = Error> {
     future::result(Ok(HttpResponse::Ok().json(creq.limits)))
 }
+
+/** Returns a status message indicating the state of the current server
+ *
+ * * version - Version information
+ * * state_data - Is the app state data present ("good" | "fail")
+ * * database - Database connection and status ("good" | "warn" | "fail")
+ * * status - Overall system status ("good" | "fail")
+ */
+pub fn heartbeat(hb: HeartbeatRequest) -> impl Future<Item = HttpResponse, Error = Error> {
+    use crate::server::ServerState;
+
+    let mut checklist = HashMap::new();
+
+    checklist.insert(
+        "version",
+        Value::String(env!("CARGO_PKG_VERSION").to_owned()),
+    );
+    checklist.insert("status", Value::String("good".to_owned()));
+    checklist.insert("state_data", Value::String("good".to_owned()));
```

Comment:
I don't think there's much value in reporting on ServerState. In practice it should always be accessible, only not if we've really badly misconfigured something and that surely would be caught much earlier than a healthcheck call.

Switching top level "status" to "fail" in that case is fine/accurate, though

Reply:
I'm kind of on the fence. I agree that "state_data" isn't super high value, until the server is returning fails and the db info looks correct and we're wondering what the heck is going on.